src/model/guild/member.rs

```diff
@@ -390,7 +390,8 @@ impl Member {
         #[cfg(feature = "cache")]
         {
             if let Some(cache) = cache_http.cache() {
-                if let Some(guild) = cache.guilds.get(&self.guild_id) {
+                let guild_ = cache.guilds.get(&self.guild_id).as_deref().cloned();
+                if let Some(guild) = guild_ {
```

Comment:
```suggestion
                let lookup = cache.guild(self.guild_id).as_deref().cloned();
                if let Some(guild) = lookup {
```
This is the correct fix, as it also prevents the chance of deadlocking (since `cache.guilds.get()` returns a different type than `cache.guild()`)